src/routes/Profile/EducationHistory/loader.ts

```diff
@@ -1,7 +1,14 @@
 import { client } from '@/api/client';
+import { authProvider } from '@/providers';
 import { LoaderFunction } from 'react-router-dom';
 
 export default (async ({ request }) => {
-  const { data } = await client.GET('/api/profiili/koulutuskokonaisuudet', { signal: request.signal });
-  return data;
+  try {
+    if (authProvider.loginState === 'loggedIn') {
+      const { data } = await client.GET('/api/profiili/koulutuskokonaisuudet', { signal: request.signal });
+      return data;
+    }
+  } catch (error) {
+    authProvider.expireSession();
```

Comment:
I think it's a bit too harsh, since there might be all kinds of server errors. I'm wondering if this provider could be used in `client.ts`. Maybe create some error handling middleware that checks the HTTP response error code. I'd try a check like this:
```
onResponse(res) {
  if (authProvider.status === 'loggenId' && res.status === 403) {
    // Was logged in but now we get unauthorized error -> session expired?
      authProvider.expireSession();
  }
}
```